### PR TITLE
Add `closest_points_between_segments()` basis path tests for Geometry2D

### PR DESCRIPTION
**Pull request by alexbilledeaux**
New tests for the closest_points_between_segments method in Geometry2D, based on basis path testing.

### CODE REVIEW

**Comment by akien-mga** [comment]
There are some style issues, see the diff from CI: https://github.com/godotengine/godot/pull/48652/checks?check_run_id=2559024734

See [Code style guidelines](https://docs.godotengine.org/en/latest/community/contributing/code_style_guidelines.html) for details, there are instructions for running `clang-format` locally to validate your changes.

File: tests/test_geometry_2d.h [hunk 1]
```diff
@@ -161,17 +161,67 @@ TEST_CASE("[Geometry2D] Closest point to uncapped segment") {
...
+		CHECK(c2.is_equal_approx(Vector2(2.5, 2.5)));
+	}
+
+	SUBCASE("[Geometry2D] Closest point on s2 trajectory is below [0,1]") {
+		Geometry2D::get_closest_points_between_segments(Vector2(-20,-20), Vector2(-10,-40), Vector2(10,25), Vector2(25,40), c1, c2);
+		CHECK(c1.is_equal_approx(Vector2(-20, -20)));
+		CHECK(c2.is_equal_approx(Vector2(10, 25)));
+	}
+
+	SUBCASE("[Geometry2D] s2 degenerates to a point") {
```
Comment (Xrayez) [inline tests/test_geometry_2d.h:195]: You probably mean:

```suggestion
	SUBCASE("[Geometry2D] c2 degenerates to a point") {
```

Same for other occurrences.

Also capitalize subcase names whenever possible.

File: tests/test_geometry_2d.h [hunk 2]
```diff
@@ -161,17 +161,67 @@ TEST_CASE("[Geometry2D] Closest point to uncapped segment") {
...
+		CHECK(c1.is_equal_approx(Vector2(1.5, 1.5)));
+		CHECK(c2.is_equal_approx(Vector2(3, 3)));
+	}
+
+	SUBCASE("[Geometry2D] s1 degenerates to a point") {
+		Geometry2D::get_closest_points_between_segments(Vector2(1,1), Vector2(1,1), Vector2(2,2), Vector2(4,4), c1, c2);
+		CHECK(c1.is_equal_approx(Vector2(1, 1)));
+		CHECK(c2.is_equal_approx(Vector2(2, 2)));
+	}
+	// End Basis Path Testing suite
```
Comment (Xrayez) [inline tests/test_geometry_2d.h:206]: In that case, I think it's worth breaking those subcases down into a separate `TEST_CASE`s.

File: tests/test_geometry_2d.h (hunk 1 above)
Comment (alexbilledeaux) [reply tests/test_geometry_2d.h:195]: Noted about the subcase names. Thanks! I was using s2 to mean "segment 2", which is definitely unclear. Would it make sense to change all instances to "segment 2"?

Comment (Xrayez) [reply tests/test_geometry_2d.h:195]: "First/Second segment" should be fine as well.

File: tests/test_geometry_2d.h (hunk 2 above)
Comment (Xrayez) [reply tests/test_geometry_2d.h:206]: Well, after thinking about it, current way of using subcases looks fine to be tbh, I just don't see much value in separating those subcases via comments like that. The problem of testing all control paths should be handled by coverage reports (which we don't have implemented for now, unfortunately).

**Review by Calinou — APPROVED** [review]
Tested locally (rebased on top of `master` https://github.com/godotengine/godot/commit/83b916bb00eb8cac4584143a80f5cfad7ce1e908), it works as expected. Code looks good to me.

I force-pushed a rebased version which also removed a few existing tests, as I *think* they are made redundant by this PR (please double-check this).

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

And sorry for the delay getting this reviewed and merged.

**Pull request merged**